src/Cake.Common/Tools/VSWhere/VSWhereAliases.cs

```diff
@@ -0,0 +1,246 @@
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.Latest")]
+        public static string VSLatest(this ICakeContext context, VSWhereLatestSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            var latest = new VSWhereLatest(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools);
+            return latest.Latest(settings);
+        }
+
+        /// <summary>
+        /// Gets all Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSAll();
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("All")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.All")]
+        public static IEnumerable<string> VSAll(this ICakeContext context)
+        {
+            var settings = new VSWhereAllSettings();
+            return VSAll(context, settings);
+        }
+
+        /// <summary>
+        /// Gets all Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSAll(new VSWhereAllSettings { Requires = "'Microsoft.Component.MSBuild" });
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("All")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.All")]
+        public static IEnumerable<string> VSAll(this ICakeContext context, VSWhereAllSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            var all = new VSWhereAll(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools);
+            return all.All(settings);
+        }
+
+        /// <summary>
+        /// Gets Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <returns>The Visual Studio installation paths.</returns>
+        /// <example>
+        /// <code>
+        ///     var latestInstallationPaths = VSProducts();
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Product")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.Product")]
+        public static IEnumerable<string> VSProducts(this ICakeContext context)
```

Comment:
This fails for we with exit code 87 (`Error 0x57: Argument required for: products`)
Docs state
```
One or more products to find. Defaults to Community, Professional, and Enterprise.
Specify "*" by itself to search all product instances installed.
``` 
Verified by calling ` -products * -property installationPath -nologo` it doesn't get this error, so if no value specified it should fall back to `*`